cmd/management-console/shared/main.go

```
@@ -177,44 +177,61 @@ func serveGet(w http.ResponseWriter, r *http.Request) {
 func serveUpdate(w http.ResponseWriter, r *http.Request) {
 	logger := log15.New("route", "update")
 
+	httpError := func(error_ error, code string) {
+		if error_ == nil {
+			error_ = errors.New("unknown")
+		}
+		logger.Error("Error updating critical configuration", "error", error_)
+		err := json.NewEncoder(w).Encode(struct {
+			Error string `json:"error"`
+			Code  string `json:"code"`
+		}{
+			Error: error_.Error(),
+			Code:  code,
+		})
+		if err != nil {
+			http.Error(w, "err", http.StatusInternalServerError)
+		}
+	}
```

I would switch to using the implementation I suggested over Slack here for the following reasons:

- `http.Error` takes an error string -- not an `error` type -- so I think taking one here would also make sense.
- Underscores in variable names is uncommon in my experience, so `error_` sticks out like a sore thumb for me.
- I think it is better if this function is not responsible for logging, as that makes this function have a less clear separation of concerns.
- You probably _do_ want to ignore the error explicitly from `json.NewEncoder().Encode` because it could have written a partial response -- there isn't anything good to do in this situation really and the most common case would be the other end hung up.

```go
func httpError(w http.ResponseWriter, error, code string) {
    _ = json.NewEncoder(w).Encode(struct{
        Error string `json:"error"`
        Code string `json:"code"`
    }{
        Error: error,
        Code: code,
    })
}
```